sycl/doc/extensions/experimental/sycl_ext_matrix/sycl_ext_oneapi_matrix.asciidoc

```diff
@@ -0,0 +1,926 @@
+fashion. `stride` describes the number of elements between consecutive
+rows for the row major layout, or between columns for the column major layout.
+
+
+==== Multiply and Add
+
+```c++
+namespace sycl::ext::oneapi::experimental::matrix {
+
+template <typename Group, typename Ta, typename Tb, typename Tc,
+          std::size_t M, std::size_t K, std::size_t N, layout LayoutA, layout
+          LayoutB, typename Td = Tc>
+joint_matrix<Group, Td, use::accumulator, M, N, layout::dynamic>
+joint_matrix_mad(Group g,
+    const joint_matrix<Group, Ta, use::a, M, K, LayoutA> &A,
+    const joint_matrix<Group, Tb, use::b, K, N, LayoutB> &B,
+    const joint_matrix<Group, Tc, use::accumulator, M, N, layout::dynamic> &C);
+
+} // namespace sycl::ext::oneapi::experimental::matrix
+```
+The matrix multiply and add function performs the multiply operation
+on the matrices `A` and `B`, accumulates the result with `C` and returns
+the result.
+
+Each device supports only certain combinations of types for the `A`,
+`B`, and `C` matrices. The application must use the query operations
+(defined below) to ensure that the combination of types is supported
+on the device where the kernel calling `joint_matrix_mad` runs.
+
+==== Fill (Initialization)
+Unlike `joint_matrix_load` that assumes that all the matrices are
+directly loaded from memory, `joint_matrix_fill`  makes it possible to
+multiply a matrix which is not directly loaded from memory but rather
+initialized directly in the register. Note that the value type `Tv`
+must be convertible to the matrix elements type `T`.
+
+```c++
+namespace sycl::ext::oneapi::experimental::matrix {
+
+template <typename Group, typename T, size_t Rows, size_t Cols,
+          use Use, layout Layout, typename Tv>
+void joint_matrix_fill(Group g, joint_matrix<Group, T, Use, Rows,
+          Cols, Layout> &m, Tv v);
+
+} // namespace sycl::ext::oneapi::experimental::matrix
+```
+
+==== Copy
+```c++
+namespace sycl::ext::oneapi::experimental::matrix {
+
+template <typename Group, typename T1, typename T2, size_t Rows,
+          size_t Cols, use Use1, use Use2, layout Layout> 
+void joint_matrix_copy(Group g,
+                       joint_matrix<Group, T1, Use1, Rows, Cols, Layout> &src,
+                       joint_matrix<Group, T2, Use2, Rows, Cols, Layout> &dest);
+
+} // namespace sycl::ext::oneapi::experimental::matrix
+```
+This function copies `Rows x Cols` elements of type `T1` from joint
+matrix `src` to `Rows x Cols` elements of type `T2` of joint matrix
+`dest`. The two matrices must have the same scope, shape, and
+layout. Use and type can be different so this function converts
+between different `use` of matrices. The application must ensure that
+the resulting matrix type is supported on the device where the kernel
+using this `joint_matrix` runs. The query functions (defined below)
+may be used to determine the set of combinations that are supported on
+a device.
```

Comment:
I think you do not need any of this description:

> The application must ensure that the resulting matrix type is supported on the device where the kernel
using this `joint_matrix` runs. The query functions (defined below) may be used to determine the set of combinations that are supported on a device.

We already say that the matrix `T` type must be supported by the device in the section titled "Element Type".